packages/beacon-state-transition/src/util/genesis.ts

```diff
@@ -89,7 +89,7 @@ export function getGenesisBeaconState(
   state.eth1Data = genesisEth1Data;
   state.randaoMixes = randaoMixes;
 
-  return createCachedBeaconState(config, state);
+  return createCachedBeaconState(config, state, {skipSyncCommitteeCache: true});
```

Comment:
Why is this added here? Don't we want the cache populated if the chain is initialized with ALTAIR_FORK_EPOCH=0 ?

Reply:
We need a CachedBeaconState to run processDeposit() which uses various caches. However at this point the state's syncCommittees are not known. This function can be called by:
  - 1. genesis spec tests: Don't care about the committee cache
  - 2. genesis builder: Only supports starting from genesis at phase0 fork
  - 3. interop state: Only supports starting from genesis at phase0 fork

So it's okay to skip syncing the sync committee cache here and expect it to be populated latter when the altair fork happens for cases 2, 3.

Reply:
Note that in current master the sync committee caches are populated with wrong data, because the logic wasn't checking that the pubkeys in the state's syncCommittee actually existed

Reply:
maybe we just need to drop a comment why we skip computing sync committees here, master works the same way (and it misses the comment that make me search for a while)

Reply:
Already added the comment